src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/authentication/OrderlyWebAuthenticationConfig.kt

```diff
@@ -1,46 +1,65 @@
 package org.vaccineimpact.orderlyweb.security.authentication
 
 import org.pac4j.core.client.IndirectClient
-import org.pac4j.core.profile.CommonProfile
 import org.pac4j.core.credentials.Credentials
+import org.pac4j.core.profile.CommonProfile
+import org.slf4j.LoggerFactory
 import org.vaccineimpact.orderlyweb.db.AppConfig
 import org.vaccineimpact.orderlyweb.db.Config
 import org.vaccineimpact.orderlyweb.security.clients.*
 
-open class AuthenticationConfig(val appConfig: Config = AppConfig())
+interface AuthenticationConfig
 {
+    val allowAnonUser: Boolean
+    fun getConfiguredProvider(): AuthenticationProvider
+    fun getAuthenticationIndirectClient(): IndirectClient<out Credentials, out CommonProfile>
+    fun getAuthenticationDirectClient(): OrderlyWebTokenCredentialClient
+}
 
-    open fun getConfiguredProvider(): AuthenticationProvider {
+class OrderlyWebAuthenticationConfig(val appConfig: Config = AppConfig()) : AuthenticationConfig
+{
+    override val allowAnonUser: Boolean
+        get()
```

Comment:
Changing this from a lazily evaluated val to an always evaluated val fixed the tests. Sadly this does not solve the mystery of why it worked just fine when running them locally but I guess I can live with not knowing.

Reply:
:woman_shrugging: